### PR TITLE
Feature/ZCS -10191&10146 - Reset Password Link and Forgot Password Changes

### PR DESCRIPTION
**Pull request by sreejan-chowdhury**
This PR is the combination of ZCS -10191 and ZCS -10146 done in ZimbraX.

### CODE REVIEW

File: soap/src/java/com/zimbra/soap/admin/message/ResetAccountPasswordRequest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2020 Synacor, Inc.
```
Comment (dasiyogesh) [inline soap/src/java/com/zimbra/soap/admin/message/ResetAccountPasswordRequest.java:4]: change the year to 2021 here and other files

File: store/src/java/com/zimbra/cs/service/admin/ResetAccountPassword.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        Account account = prov.get(AccountBy.fromString(key), value, zsc.getAuthToken());
+
+        // prevent directory harvest attack, mask no such account as permission denied
+        if (account == null)
+            throw ServiceException.PERM_DENIED("can not access account");
+
+        checkAccountRights(zsc, account);
+        account.refreshUserCredentials();
+
+        if(account.getFeatureResetPasswordStatus().equals(FeatureResetPasswordStatus.suspended)) {
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/service/admin/ResetAccountPassword.java:67]: add space between if & (, here and below

Comment (sreejan-chowdhury) [reply store/src/java/com/zimbra/cs/service/admin/ResetAccountPassword.java]: done

File: soap/src/java/com/zimbra/soap/admin/message/ResetAccountPasswordRequest.java (hunk 1 above)
Comment (sreejan-chowdhury) [reply soap/src/java/com/zimbra/soap/admin/message/ResetAccountPasswordRequest.java]: done

**Review by chinmay15 — APPROVED** [review]
LGTM

**Comment by chinmay15** [comment]
merged in develop

commit 502c6be7b05b88996ad10c6e388018c3def7dbc1 (HEAD -> develop, origin/develop, origin/HEAD)
Merge: 4058af2249 ca2df3e452
Author: Chinmay Pathak <15.chinmay@gmail.com>
Date:   Tue Mar 23 19:22:04 2021 +0530

    Merge branch 'feature/ZCS-10191-10146' into develop

**Pull request merged**